File: viet_nam_trip/app/Models/goi_du_lich.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class goi_du_lich extends Model
{
    use HasFactory;
    use SoftDeletes;

    public $timestamps = true;

    protected $fillable = [
        'loai_id',
        'ten',
        'hinh_goi_du_lich',
        'noi_khoi_hanh',
        'thoi_gian_tap_trung',
        'so_nguoi_toi_da',
        'gio_khoi_hanh',
        'ngay_khoi_hanh',
        'gia_nguoi_lon',
        'gia_tre_em',
        'gia_tre_nho',
        'so_ngay',
        'so_dem',
        'not_compo',
        'compo',
        'thong_tin_dich_vu',
        'dieu_kien_ap_dung',
        'trang_thai',
        'noi_bat',
    ];

    public function loai_goi_du_lich(){
        return $this->hasMany('App\Models\loai_goi_du_lich', 'id', 'loai_id');
    }

    public function lich_dat(){
        return $this->hasMany('App\Models\lich_dat', 'id', 'goi_du_lich_id');
    }

    public function lich_trinh(){
        return $this->hasMany('App\Models\lich_trinh', 'id', 'goi_du_lich_id');
    }

    public function phieu_dat(){
        return $this->hasMany('App\Models\phieu_dat', 'goi_du_lich_id', 'id');
    }
}

File: dulich/lib/Models/goi_dulich.dart
class GoiDuLich {
  final int id;
  final String ten;
  final String hinh_goi_du_lich;
  final String noi_khoi_hanh;
  final String ngay_khoi_hanh;
  final int gia_nguoi_lon;
  final int gia_tre_em;
  final String thong_tin_dich_vu;

  GoiDuLich(
    this.id,
    this.ten,
    this.hinh_goi_du_lich,
    this.noi_khoi_hanh,
    this.ngay_khoi_hanh,
    this.gia_nguoi_lon,
    this.gia_tre_em,
    this.thong_tin_dich_vu,
  );
  GoiDuLich.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        ten = json['ten'],
        hinh_goi_du_lich = json['hinh_goi_du_lich'],
        noi_khoi_hanh = json['noi_khoi_hanh'],
        ngay_khoi_hanh = json['ngay_khoi_hanh'],
        gia_nguoi_lon = json['gia_nguoi_lon'],
        gia_tre_em = json['gia_tre_em'],
        thong_tin_dich_vu = json['thong_tin_dich_vu'];
}
